externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/SparkOperation.scala

```diff
@@ -250,7 +250,7 @@ abstract class SparkOperation(session: Session)
           } else {
             val taken = iter.take(rowSetSize)
             RowSet.toTRowSet(
-              taken.toSeq.asInstanceOf[Seq[Row]],
+              taken.toIndexedSeq.asInstanceOf[Seq[Row]],
```

Comment:
Please leave some comments here to explain why we should use index-access-friendly data structure here